modules/data_models.py

```diff
@@ -0,0 +1,363 @@
+        """
+        Initializes the members name, year and path and
+        calls populate to fill days_data list
+        Args:
+            year(str or int):   Value containing 4 digit year e.g '2004'
+            month(str): Value containing 3 character month name e.g 'Feb', 'Mar'
+            path(str):  Value containing path to weather files e.g 'weatherfiles/'
+        """
+        self.days_data = []
+        self.name = month
+        self.year = year
+        self.path = path
+        self.populate()
+
+    def populate(self):
+        """
+        Initializes the days_data list with DayData objects for each day of the month
+        Returns:
+            None
+        """
+
+        month_data = next(read_data(f"{self.year}_{self.name[0:3]}", self.path))
+        for line in month_data:
+            day_data = DayData(parse_line(line))
+            self.days_data.append(day_data)
+
+    def get_max_extremes_with_date(self):
+        """
+        Returns the max highest temperature, min lowest temperature and max humidity
+        for the whole month with respective dates
+        Returns:
+            (list or None): list containing 3 elements
+                            [
+                                [max_temperature(int), date(str)],
+                                [min_temperature(int), date(str)],
+                                [max_humidity(int), date(str)],
+                            ]
+                            Or
+                            None if no data exists for the month
+        """
+        max_temperature = {"value": -1000, "date": ""}
+        min_temperature = {"value": 1000, "date": ""}
+        max_humidity = {"value": -1000, "date": ""}
+        for day_data in self.days_data:
+            date = day_data.get_date()
+
+            max_arg2 = {"value": day_data.get_max_temperature(), "date": date}
+            max_temperature = max(
+                max_temperature, max_arg2, key=lambda x: x["value"] or -1000
+            )
+
+            min_arg2 = {"value": day_data.get_min_temperature(), "date": date}
+            min_temperature = min(
+                min_temperature, min_arg2, key=lambda x: x["value"] or 1000
+            )
+
+            max_arg2 = {"value": day_data.get_max_humidity(), "date": date}
+            max_humidity = max(
+                max_humidity, max_arg2, key=lambda x: x["value"] or -1000
+            )
+        return [
+            max_temperature,
+            min_temperature,
+            max_humidity,
+        ]
+
+    def get_averages(self):
+        """
+        Returns the averages highest_temperature, lowest_temperature and mean_humidity
+        Returns:
+            (list or None): list containing rounded averages
+                            [
+                                avg_highest_temperature(str),
+                                avg_lowest_temperature(str),
+                                avg_mean_humidity(str)
```

Comment:
Similar to the above comments, We should divide it into multiple independent functions that we can call independently to populate a data matrix.

Reply:
Done